react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+class Stopwatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = ({
+      isRunning: false,
+      timeElapsed: 0
+    });
+    this.incrementor = null;
+    this.handleStartTimer = this.handleStartTimer.bind(this);
+    this.addToTimer = this.addToTimer.bind(this);
+    this.handleStopTimer = this.handleStopTimer.bind(this);
+  }
+
+  addToTimer() {
+    this.setState(
+      { timeElapsed: this.state.timeElapsed + 1 });
+  }
+
+  handleStartTimer() {
+    this.incrementor = setInterval(this.addToTimer, 1000
+    );
+    this.setState({
+      isRunning: true
+    });
+  }
+
+  handleStopTimer() {
+    clearInterval(this.incrementor);
+    this.setState({
+      isRunning: false,
+      timeElapsed: 0
+    });
+  }
```

Comment:
The stopwatch should be able to pause and resume without losing the amount of seconds passed. However, if the stopwatch is paused, then the user should be able to reset it by clicking the watch face.